openedx/core/djangoapps/user_authn/views/tests/test_filters.py

```diff
@@ -0,0 +1,208 @@
+            },
+        },
+    )
+    def test_register_filter_prevent_registration(self):
+        """
+        Test prevent the user's registration through a pipeline step.
+
+        Expected result:
+            - PreRegisterFilter is triggered and executes TestStopRegisterPipelineStep.
+            - The user's registration stops.
+        """
+        response = self.client.post(self.url, self.user_info)
+
+        self.assertEqual(status.HTTP_403_FORBIDDEN, response.status_code)
+
+
+@skip_unless_lms
+class LoginFiltersTest(UserAPITestCase):
+    """
+    Tests for the Open edX Filters associated with the user login process.
+
+    This class guarantees that the following filters are triggered during the user's login:
+
+    - PreLoginFilter
+    """
+
+    def setUp(self):  # pylint: disable=arguments-differ
+        super().setUp()
+        self.user = UserFactory.create(
+            username="test",
+            email="test@example.com",
+            password="password",
+        )
+        self.user_profile = UserProfileFactory.create(user=self.user, name="Test Example")
+        self.url = reverse('login_api')
+
+    @override_settings(
+        OPEN_EDX_FILTERS_CONFIG={
+            "org.openedx.learning.student.login.requested.v1": {
+                "pipeline": [
+                    "openedx.core.djangoapps.user_authn.views.tests.test_filters.TestLoginPipelineStep",
+                ],
+                "fail_silently": False,
+            },
+        },
+    )
+    def test_login_filter_executed(self):
+        """
+        Test whether the student login filter is triggered before the user's
+        login process.
+
+        Expected result:
+            - PreLoginFilter is triggered and executes TestLoginPipelineStep.
+            - The user's profile is updated.
+        """
+        data = {
+            "email": "test@example.com",
+            "password": "password",
+        }
+
+        self.client.post(self.url, data)
+
+        user = User.objects.get(username=self.user.username)
+        self.assertDictEqual({"logged_in": True}, user.profile.get_meta())
+
+    @override_settings(
+        OPEN_EDX_FILTERS_CONFIG={
+            "org.openedx.learning.student.login.requested.v1": {
+                "pipeline": [
+                    "openedx.core.djangoapps.user_authn.views.tests.test_filters.TestStopLoginPipelineStep",
+                ],
+                "fail_silently": False,
+            },
+        },
+    )
+    def test_login_filter_prevent_login(self):
+        """
+        Test prevent the user's login through a pipeline step.
+
+        Expected result:
+            - PreLoginFilter is triggered and executes TestStopLoginPipelineStep.
+            - Test prevent the user's login through a pipeline step.
+        """
+        data = {
+            "email": "test@example.com",
+            "password": "password",
+        }
+
+        response = self.client.post(self.url, data)
```

Comment:
Could you please put in a test that returns this successfully when the filter is not running? That way, we can be absolutely sure that it's the filter causing the 400 error, and not that `data` is somehow being malformed in the test itself.